camera: move to REQUEST_MESSAGE

julianoes opened the pull request:
Closes #298.

hamishwillee commented:
Fantastic thanks @julianoes . I did an update to tidy up the rendered mermaid diagrams and moved the migration notes to be a section and the end with a warning up the top. I.e. these need to be visible and clearly "just for GCS/Camera server". 

One point of discussion, is that you referred to GCS and camera servers needing to support both formats. I have suggested that GCS and MAVLink SDKs should support both protocol versions. I see camera servers as just another camera and my understanding is that the model is that cameras update to new protocol, GCS support both.

If you're happy with this change, please merge.

hamishwillee commented:
@julianoes We chatted a bit offline about chicken/egg of cameras vs GCS need to support both systems. IMO cameras should upgrade, and only need to support one system. Ditto camera servers. I am not sure if you agreed, but if you did then please merge. And if you did not, then please fix up to match your expectation :-)

julianoes commented:
Sorry, I changed it back the way I think it should work, and removed the year estimation because it's going to be wrong I just don't know in which direction yet :joy:.

hamishwillee commented:
All good thanks! Sounds like a plan.
